<launch>
    <!-- Set use_sim_time true to generate trajectory bag file in same time domain -->
    <param name="/use_sim_time" value="true"/>

    <!-- Path of a rosbag file -->
    <arg name="bag_path" default="$(find path_recorder)/../bag/nclt_2011_09_30_drive_0027_synced"/>


    <!-- Run Fast_LIO -->
    <group>
        <rosparam command="load" file="$(find fast_lio)/config/velodyne.yaml" />
    <arg name="rviz" default="true" />

    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num" type="int" value="4"/>
    <param name="max_iteration" type="int" value="3" />
    <param name="filter_size_surf" type="double" value="0.5" />
    <param name="filter_size_map" type="double" value="0.5" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="1" />
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" /> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
    </group>
    <node pkg="tf2_ros" type="static_transform_publisher"
          name="world_vio_init_pub" args="0 0 0 1 0 0 0 world slam_init" />  
    </group>


	<!-- Remap point cloud topic -->
	<remap from="points_raw" to="velodyne_points" />
	<!-- Play nclt rosbag file -->
    <node pkg="rosbag" type="play" name="nclt" args="--clock $(arg bag_path)/nclt.bag  -u 100" output="log" />

	<!-- Launch path_recorder to save a trajectory from FAST_LIO -->
    <node pkg="path_recorder" type="path_recorder" name="fast_recorder" output="screen">
        <param name="algorithm_name" type="string" value="fast_lio" />
        <param name="path_topic" type="string" value="/path" />
        <param name="save_to" type="string" value="$(arg bag_path)" />
    </node>
</launch>
